lib/SciMLCompat/src/enzyme.jl

```diff
@@ -0,0 +1,278 @@
+""" 
+This module wraps some of the functions of the IncompressibleNavierStokes module
+in order to make them compatible with Enzyme AD.
+
+There are also some tests to check if the redefined functions are faster than the original ones and if they are differentiable.
+"""
+
+using IncompressibleNavierStokes 
+using SparseArrays
+using LinearAlgebra
+using KernelAbstractions
+using Enzyme
+# runtimeActivity is required to tell Enzyme that all temporary variables should be active
+Enzyme.API.runtimeActivity!(true)
+
+struct Offset{D} end
+@inline (::Offset{D})(α) where {D} = CartesianIndex(ntuple(β -> β == α ? 1 : 0, D))
+
+"""
+    _get_enz_bc_u!(_backend, setup)
+
+This function is used to precompile the function that applies the boundary conditions on u.
+It wraps everything such that the only thing that has to be passed from outside is the u array.
+This makes it easy for Enzyme to differentiate the function.
+"""
+function _get_enz_bc_u!(_backend, setup)
```

Comment:
exclamation marks signal that the function edits arguments, its not the case here I think